File: 1732-minimum-one-bit-operations-to-make-integers-zero/minimum-one-bit-operations-to-make-integers-zero.py
class Solution:
    def minimumOneBitOperations(self, n: int) -> int:
        ans = 0
        f = 0
        for i in range(31, -1, -1):
            if ((n >> i) & 1) == 1:
                if f == 0:
                    ans = ans + ((1 << (i + 1))) - 1
                    f = 1
                else:
                    ans = ans - ((1 << (i + 1)) - 1)
                    f = 0
        return ans

File: 1732-minimum-one-bit-operations-to-make-integers-zero/minimum-one-bit-operations-to-make-integers-zero.java
class Solution {
    public int minimumOneBitOperations(int n) {
        int ans = 0;
        int mul = 1;
        while (n > 0) {
            int shift = (int) Math.floor((Math.log(n) / Math.log(2)));
            ans += ((1 << (shift + 1)) - 1) * mul;
            n -= 1 << shift;
            mul *= -1;
        }
        
        return ans;
    }
}